### ISSUE TITLE
Upgrading to v2.0.790.0 on Windows is removing the Environment registry key (proxy)

### ISSUE THREAD

**Issue by PaulAtkins** [aws/amazon-ssm-agent#54]
Hey,

It seems like when calling `AWS-UpdateSSMAgent` on Windows and using the default parameters, it's upgrading to Agent version `v2.0.790.0` and the SSM Command Doc completes successfully.

But the `Environment` key under `HKLM:\SYSTEM\CurrentControlSet\Services\AmazonSSMAgent` is being removed, so when using a proxy, the upgraded Agent can no longer connect to the SSM service.

### Steps to reproduce:

1. Launch a `Windows_Server-2016-English-Full-Base-2017.05.22 (ami-c2c3a2a2)` instance, with the following UserData (updating `hostname:port` to the correct value) to configure the SSM Agent to use a proxy and restart the service.

```
$serviceKey = "HKLM:\SYSTEM\CurrentControlSet\Services\AmazonSSMAgent"
$keyInfo = (Get-Item -Path $serviceKey).GetValue("Environment")
$proxyVariables = @("http_proxy=hostname:port", "no_proxy=169.254.169.254")

If($keyInfo -eq $null)
{
New-ItemProperty -Path $serviceKey -Name Environment -Value $proxyVariables -PropertyType MultiString -Force
}
else
{
Set-ItemProperty -Path $serviceKey -Name Environment -Value $proxyVariables
}

Restart-Service AmazonSSMAgent 
```
From : http://docs.aws.amazon.com/systems-manager/latest/userguide/ssm-agent.html#sysman-install-ssm-win

2. Confirm the Agent appears under `Managed Instances` as Online

3. Run the `AWS-UpdateSSMAgent` with the default parameters

4. Check the Command completed successfully and upgraded the Agent. Example output:
> Updating amazon-ssm-agent from 2.0.767.0 to latest
Successfully downloaded https://s3.us-west-2.amazonaws.com/amazon-ssm-us-west-2/ssm-agent-manifest.json
Successfully downloaded https://s3.us-west-2.amazonaws.com/amazon-ssm-us-west-2/amazon-ssm-agent-updater/2.0.790.0/amazon-ssm-agent-updater-windows-amd64.zip
Successfully downloaded https://s3.us-west-2.amazonaws.com/amazon-ssm-us-west-2/amazon-ssm-agent/2.0.767.0/amazon-ssm-agent-windows-amd64.zip
Successfully downloaded https://s3.us-west-2.amazonaws.com/amazon-ssm-us-west-2/amazon-ssm-agent/2.0.790.0/amazon-ssm-agent-windows-amd64.zip
Initiating amazon-ssm-agent update to 2.0.790.0
amazon-ssm-agent updated successfully to 2.0.790.0

### Expected Result:
- The `Environment` key is preserved
- The Agent appears under `Managed Instances` and shows the latest version `2.0.790.0` and the Ping Status is `Online`
- SSM Command Documents can still be run successfully on the Instance

### Actual Result:
- The `Environment` key is removed
- The Agent appears under `Managed Instances` and shows the original version `2.0.767.0` and the Ping Status becomes `ConnectionLost`
- SSM Command Documents fail to run on the Instance, and time out

Note: Remoting into the instance and running the same script UserData script re-adds the `Environment` key, and the Agent appears correctly under `Managed Instances` as Online with the correct version

**Comment by mmendonca3** [comment]
Hello PaulAtkins, 

Thank you for posting here. We are investigating this issue and will update about the fix in the release notes once rolled out. 

Thank you!

**Comment by PaulAtkins** [comment]
Thanks for fixing this - looks like this got fixed in release v2.0.822.0, and I'm no longer getting this problem, so closing this issue.,
